code/core/src/test/java/com/adobe/marketing/mobile/services/NetworkServiceTests.java

```diff
@@ -39,12 +46,60 @@ public class NetworkServiceTests {
 
     @Mock DeviceInfoService deviceInfoService;
 
+    @Mock AppContextService appContextService;
+
+    @Mock ConnectivityManager connectivityManager;
+
     private HttpConnectionHandler httpConnectionHandler;
 
     @Before
     public void setup() throws Exception {
         networkService = new NetworkService(currentThreadExecutorService());
         ServiceProvider.getInstance().setDeviceInfoService(deviceInfoService);
+        ServiceProvider.getInstance().setAppContextService(appContextService);
+        when(appContextService.getConnectivityManager()).thenReturn(null);
+    }
+
+    @Test(timeout = 1000L)
+    public void testConnectAsync_InternetNotAvailable() throws InterruptedException {
```

Comment:
IMO these _functional_ tests add very limited value given that the `AppContextService` and `NetworkUtils` are being mocked any way. So the only functional feature usage from the test suite is waiting for the network request related log to be made.

I recommend adding unit tests for `NetwtorkService` for a similar verification instead by ensuring that the executor service does spin up a thread if there is no network. This can be done by mocking the ExecutorService and the NetworkUtils.kt directly. That way we would not need to add the additional test dependencies.

Reply:
1. `testConnectAsync_InternetNotAvailable` mainly tests if the network callback is invoked with the null connection (`assertNull(connection);`) when the network is unavailable. It's the main contract between the Network Services and its client. This is the basic test case that we should add for sure. 
2. I agree that we can add some logic to check `ExecutorService` execution in these tests.

Reply:
For usecase 1 We can still test that the callback is invoked with `null` if we mock the network request and response below:
```
networkService.connectAsync(mockRequest, mockResponseCallback)
verify(mockResponseCallback, times).call(null)
```

For verifying that the network request is made when internet is available it is sufficient to check if the executor service submits a thread.

Reply:
I think it's no harm to verify both. I will update the PR to include those changes.  Let me know if you have other thoughts?